docs/features/incremental-generators.md

```diff
@@ -0,0 +1,485 @@
+set of transforms that will be applied to the data as it changes. Transforms are
+applied through a set of extension methods:
+
+```csharp
+public static partial class IncrementalValueSourceExtensions
+{
+    // 1 => 1 transform 
+    public static IncrementalValueSource<U> Transform<T, U>(this IncrementalValueSource<T> source, Func<T, U> func) => ...
+}
+```
+
+These extension methods allow the user to perform a series of transforms,
+conceptually somewhat similar to LINQ, over the values coming from the data
+source:
+
+```csharp
+initContext.RegisterExecutionPipeline((context) =>
+{
+    // get the additional text source
+    var additionalTexts = context.Sources.AdditionalTexts;
+
+    // apply a 1-to-1 transform on each text, which represents extracting the path
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+});
+```
+
+Note that `transformed` is similarly opaque. It represents the outcome of the
+transformation being applied to the data, but cannot be accessed directly.
+
+The transformed steps can be further transformed, and it is also valid to
+perform multiple transformations on the same input node, essentially 'splitting'
+the pipeline into multiple streams of processing.
+
+```csharp
+    // apply a 1-to-1 transform on each text, extracting the path
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+    // split the processing into two streams of derived data
+    var prefixTransform = transformed.Transform(path => "prefix_" + path);
+    var postfixTransform = transformed.Transform(path => path + "_postfixed");
+```
+
+### Batching
+
+In addition to the 1-to-1 transform shown above, there are also extension
+methods for producing and consuming batches of data. For instance a given
+transform may want to produce more than one value for each input, or want to
+view all the data in a single collected view in order to make cross data
+decisions.
+
+``` csharp
+public static partial class IncrementalValueSourceExtensions
+{
+    // 1 => many (or none)
+    public static IncrementalValueSource<U> TransformMany<T, U>(this IncrementalValueSource<T> source, Func<T, IEnumerable<U>> func) => ...
+
+    // many => 1
+    public static IncrementalValueSource<U> BatchTransform<T, U>(this IncrementalValueSource<T> source, Func<IEnumerable<T>, U> func) => ...
+
+    // many => many (or none)
+    public static IncrementalValueSource<U> BatchTransformMany<T, U>(this IncrementalValueSource<T> source, Func<IEnumerable<T>, IEnumerable<U>> func) => ...
+}
+```
+
+In our above example we could use `BatchTransform` to collect the individual
+file paths collected, and convert them into a single collection:
+
+```csharp
+    // apply a 1-to-1 transform on each text, which represents extracting the path
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+    // batch the collected file paths into a single collection
+    var batched = transformed.BatchTransform(texts => texts);
```

Comment:
I think in this and the usage below it might be more clear to make the lambda parameter name `paths` instead of `texts`

Reply:
i also think you need to show the return types here (might be good to avoid var entirely in this gist) to show what impact this actually had.